installer/go/pkg/nodejs/deviceagent.go

```diff
@@ -0,0 +1,184 @@
+package nodejs
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/flowfuse/device-agent-installer/pkg/logger"
+	"github.com/flowfuse/device-agent-installer/pkg/utils"
+)
+
+const packageName = "@flowfuse/device-agent"
+
+// InstallDeviceAgent installs the FlowFuse Device Agent with the specified version
+// to the given base directory. It requires Node.js to be already installed.
+//
+// Parameters:
+//   - version: The version of the Device Agent to install (use "latest" for the latest version)
+//   - baseDir: The base directory where Node.js is installed and where the Device Agent will be installed
+//
+// The function will:
+// 1. Check if Node.js is installed
+// 2. Install the Device Agent globally using npm with the appropriate version
+// 3. The installation runs as the service user
+//
+// Returns an error if:
+// - Node.js is not found
+// - The operating system is not supported (currently only Linux is supported)
+// - The installation process fails
+//
+// Note: For Linux, the installation uses sudo to run npm as the service user.
+func InstallDeviceAgent(version string, baseDir string) error {
+	setNodeDirectories(baseDir)
+
+	if _, err := os.Stat(nodeBinPath); os.IsNotExist(err) {
+		return fmt.Errorf("node.js not found, please restart installator script")
+	}
+
+	serviceUser := utils.ServiceUsername
+
+	packageName := packageName
+	if version != "latest" {
+		packageName += "@" + version
+	}
+
+	logger.Debug("Installing %s as user %s...", packageName, serviceUser)
+
+	var installCmd *exec.Cmd
+	npmPrefix := fmt.Sprintf("npm_config_prefix=%s", nodeBaseDir)
+	newPath, _ := setEnvPath()
+
+	// Create install command
+	switch runtime.GOOS {
+	case "linux":
+		installCmd = exec.Command("sudo", "--preserve-env=PATH", "-u", serviceUser, npmBinPath, "install", "-g", packageName)
```

Comment:
This runs the installer as the `flowfuse` users.

unfortunately, this means it has zero access to other home or protected areas where an existing flows import may live.

May need to consider running npm install as elevated user then `chown` or `setfacl` the dir for `serviceUser`